specification/_types/Errors.ts

```diff
@@ -17,56 +17,35 @@
  * under the License.
  */
 
-import { PainlessExecutionPosition } from '@global/scripts_painless_execute/types'
-import { Dictionary } from '@spec_utils/Dictionary'
-import { HttpHeaders, Id, Ids, IndexName, Uuid } from './common'
+import { Id, IndexName } from './common'
 import { integer, long } from './Numeric'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import { AdditionalProperties } from '@spec_utils/behaviors'
 
-export class ErrorCause {
-  type: string
-  reason: string
-
-  caused_by?: ErrorCause
-  shard?: integer | string
-  stack_trace?: string
-
-  root_cause?: ErrorCause[]
-
-  bytes_limit?: long
-  bytes_wanted?: long
-  column?: integer
-  col?: integer
-  failed_shards?: ShardFailure[]
-  grouped?: boolean
-  index?: IndexName
-  index_uuid?: Uuid
-  language?: string
-  licensed_expired_feature?: string
-  line?: integer
-  max_buckets?: integer
-  phase?: string
-  property_name?: string
-  processor_type?: string
+/**
+ * Cause and details about a request failure. This class defines the properties common to all error types.
+ * Additional details are also provided, that depend on the error type.
+ *
+ * @shortcut_property reason
+ */
+export class ErrorCause
+  implements AdditionalProperties<string, UserDefinedValue> {
   /**
-   * resource id
-   * @aliases resource.id
+   * The type of error
    */
-  resource_id?: Ids
+  type?: string
```

Comment:
Note: `type` is optional as not all errors return it:
```
GET /_cat/foobar
{
  "error" : "Incorrect HTTP method for uri [/_cat/foobar?pretty=true] and method [GET], allowed: [POST]",
  "status" : 405
}
```

This format is captured by the `@shortcut_property reason`: if `error` is a string, then it's equivalent to an `ErrorCause` with a `reason` only.